src/Subscriber/RewriteUnwantedPhrasesSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Subscriber;
+
+use App\Api\PullRequest\PullRequestApi;
+use App\Event\GitHubEvent;
+use App\GitHubEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Lock\LockFactory;
+
+/**
+ * There are some phrases that should be avoided in title/description. This
+ * subscriber helps rewrite those.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class RewriteUnwantedPhrasesSubscriber implements EventSubscriberInterface
+{
+    private $pullRequestApi;
+    private $lockFactory;
+
+    public function __construct(PullRequestApi $pullRequestApi, LockFactory $lockFactory)
+    {
+        $this->pullRequestApi = $pullRequestApi;
+        $this->lockFactory = $lockFactory;
+    }
+
+    public function onPullRequest(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        $action = $data['action'];
+        if (!in_array($action, ['opened', 'ready_for_review', 'edited'])) {
+            return;
+        }
+
+        if (!isset($data['pull_request'])) {
+            // Only update PullRequests
+            return;
+        }
+
+        $repository = $event->getRepository();
+        $number = $data['number'];
+
+        sleep(1); // Wait for github API to be updated
+        $lock = $this->lockFactory->createLock($repository->getFullName().'#'.$number);
+        $lock->acquire(true); // blocking. Lock will be released at __destruct
```

Comment:
This is the first time that we've needed to lock and sleep? What makes this subscriber unique in that way?

Reply:
The second time: https://github.com/symfony-tools/carsonbot/blob/a08565befe2aed5e2c730523c24f89b805e5e983/src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php#L52

Right when we open a PR we do many requests to modify the PR. Which will sometimes lead to race conditions. 

This new subscriber is very similar to the AutoUpdateTitleWithLabelSubscriber

Reply:
Ah, indeed - I had somehow missed this same code in `AutoUpdateTitleWithLabelSubscriber`. That's maybe another thing that could be centralized a bit... later ;)